GCTC-NTGC/gc-digital-talent
# Restrict skills list to only "need to have" and "nice to have" skills during an application

Resolves #4365.

Changes skill selection in the experience forms to only "need to have" and "nice to have" skills when applying to a pool. The skill selection should remain the same while viewing the form from the users profile.

Note: This PR contains 2 new strings I still need to get translated.

To test:
1. From your profile navigate to the edit page for an existing experience and the add skills section appears unchanged
2. Start a pool application
3. Navigate to an edit experience page from the new application and confirm the add skills section has changed to only include skills marked as need/nice-to-have for the pool advertisement
4. Add and remove a couple skills for the experience - confirm it works as expected.
5. Save the form and confirm everything is saved as expected
